builtin/providers/azurerm/resource_arm_loadbalancer_backend_address_pool.go

```diff
@@ -0,0 +1,207 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jen20/riviera/azure"
+)
+
+func resourceArmLoadbalancerBackendAddressPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLoadbalancerBackendAddressPoolCreate,
+		Read:   resourceArmLoadbalancerBackendAddressPoolRead,
+		Delete: resourceArmLoadbalancerBackendAddressPoolDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"loadbalancer_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"backend_ip_configurations": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"load_balancing_rules": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+		},
+	}
+}
+
+func resourceArmLoadbalancerBackendAddressPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	lbClient := client.loadBalancerClient
+
+	loadBalancer, exists, err := retrieveLoadbalancerById(d.Get("loadbalancer_id").(string), meta)
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer By ID {{err}}", err)
+	}
+	if !exists {
+		d.SetId("")
+		log.Printf("[INFO] LoadBalancer %q not found. Removing from state", d.Get("name").(string))
+		return nil
+	}
+
+	_, _, exists = findLoadBalancerBackEndAddressPoolByName(loadBalancer, d.Get("name").(string))
+	if exists {
+		return fmt.Errorf("A BackEnd Address Pool with name %q already exists.", d.Get("name").(string))
+	}
+
+	backendAddressPools := append(*loadBalancer.Properties.BackendAddressPools, expandAzureRmLoadbalancerBackendAddressPools(d))
+	loadBalancer.Properties.BackendAddressPools = &backendAddressPools
+	resGroup, loadBalancerName, err := resourceGroupAndLBNameFromId(d.Get("loadbalancer_id").(string))
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer Name and Group: {{err}}", err)
+	}
+
+	_, err = lbClient.CreateOrUpdate(resGroup, loadBalancerName, *loadBalancer, make(chan struct{}))
+	if err != nil {
+		return errwrap.Wrapf("Error Creating/Updating LoadBalancer {{err}}", err)
+	}
+
+	read, err := lbClient.Get(resGroup, loadBalancerName, "")
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer {{err}}", err)
+	}
+	if read.ID == nil {
```

Comment:
Yes, a successful api call can still potentially return an empty result. This is a guard against a panic when dereferencing